src/Build/BuildCheck/Infrastructure/BuildCheckCentralContext.cs

```diff
@@ -219,15 +220,26 @@ private void RunRegisteredActions<T>(
             // Here we might want to check the configPerRule[0].EvaluationsCheckScope - if the input data supports that
             // The decision and implementation depends on the outcome of the investigation tracked in:
             // https://github.com/orgs/dotnet/projects/373/views/1?pane=issue&itemId=57851137
-
-            BuildCheckDataContext<T> context = new BuildCheckDataContext<T>(
-                checkCallback.Item1,
-                checkContext,
-                configPerRule,
-                resultHandler,
-                checkData);
-
-            checkCallback.Item2(context);
+            try
+            {
+                BuildCheckDataContext<T> context = new BuildCheckDataContext<T>(
+                    checkCallback.Item1,
+                    checkContext,
+                    configPerRule,
+                    resultHandler,
+                    checkData);
+
+                checkCallback.Item2(context);
```

Comment:
Let's scope the try/catch just to the calling of callback - the creation of the context should not throw.

We should as well deregister the check upon it experiencing unhandled error. The `BuildCheckCentralContext` already has a convenient callback for that (`_removeThrottledChecks`) - it just need to be adjusted to take the reason message for the deregistration - and then the same callbeck can be used to remove throttled checks (`checkCallback.Item1.IsThrottled == true`) and checks that receive exception here